sql/core/src/test/scala/org/apache/spark/sql/ApproximatePercentileQuerySuite.scala

```diff
@@ -279,4 +282,10 @@ class ApproximatePercentileQuerySuite extends QueryTest with SharedSQLContext {
       checkAnswer(query, expected)
     }
   }
+
+  test("SPARK-24013: unneeded compress can cause performance issues with sorted input") {
+    failAfter(20 seconds) {
+      assert(sql("select approx_percentile(id, array(0.1)) from range(10000000)").count() == 1)
```

Comment:
When you do .count(), column pruning removes the approx_percentile from the query, so the test does not execute approx_percentile.

Reply:
nice catch, thanks, I started using collect during my tests than I moved to count but it was a mistake, I am fixing it, thanks.